src/LinuxTracing/ThreadStateManager.cpp

```diff
@@ -131,23 +137,29 @@ std::optional<ThreadStateSlice> ThreadStateManager::OnSchedSwitchIn(uint64_t tim
   slice.set_wakeup_reason(open_state.wakeup_reason);
   slice.set_wakeup_tid(open_state.wakeup_tid);
   slice.set_wakeup_pid(open_state.wakeup_pid);
+  if (open_state.wait_for_callstack) {
+    slice.set_callstack_status(orbit_grpc_protos::ThreadStateSlice::kWaitingForCallstack);
```

Comment:
The comment above says that there are some cases where `open_state.state != ThreadStateSlice::kRunnable`. Would this callstack be from `wakeup_tid` in that case? I assume everything already fits to place?

Reply:
My take on this would be, there is no sched_wakeup event happening, so the slice produced here is either a runnable slice (with the callstack coming from wakeup_tid) or the sleeping slice (with the callstack coming from the thread itself).

I don't see why it should not work here.

Reply:
Makes sense. I was wondering if I had gotten everything correctly.